# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1403436146 14400
#      Sun Jun 22 07:22:26 2014 -0400
# Branch stable
# Node ID 9a70705dc61614e9ab1dad182d0ee465326e1956
# Parent  9d185537e5d12ba366f78dd06c9c4b01b2104c86
Check for name and type keyword to return as an octave ascii data file (Bug #42586)

* libinterp/corefcn/load-save.cc
  (get_file_format): attempt to get name and then type keyword to decide if is an Octave ascii file.

diff --git a/libinterp/corefcn/load-save.cc b/libinterp/corefcn/load-save.cc
--- a/libinterp/corefcn/load-save.cc
+++ b/libinterp/corefcn/load-save.cc
@@ -260,19 +260,20 @@ get_file_format (std::istream& file, con
               file.seekg (0, std::ios::beg);
               retval = LS_MAT5_BINARY;
             }
           else
             {
               file.clear ();
               file.seekg (0, std::ios::beg);
 
-              std::string tmp = extract_keyword (file, "name");
+              std::string name_val = extract_keyword (file, "name");
+              std::string type_val = extract_keyword (file, "type");
 
-              if (! tmp.empty ())
+              if (name_val.empty () != true && type_val.empty () != true)
                 retval = LS_ASCII;
               else
                 {
                   file.clear ();
                   file.seekg (0, std::ios::beg);
 
                   // FIXME: looks_like_mat_ascii_file does not check to see
                   // whether the file contains numbers.  It just skips comments
